airbyte-integrations/connectors/source-posthog/source_posthog/components.py

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Any, Mapping, MutableMapping, Optional
+
+from airbyte_cdk.sources.declarative.stream_slicers import SubstreamSlicer
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice, StreamState
+from airbyte_cdk.sources.declarative.requesters.request_options.interpolated_request_options_provider import (
+    InterpolatedRequestOptionsProvider,
+)
+
+
+@dataclass
+class PosthogIncrementalSlicer(SubstreamSlicer):
```

Comment:
could this be replaced with a CartesianStreamSlicer on top of a DatetimeStreamSlicer and a SubstreamSlicer?